### PR TITLE
fix(alerts): after dismissing alert are not removing from DOM #3608

### LINKED ISSUE #3608: fix(alerts): after dismissing alert are not removing from DOM
### WTR
- Go to Alerts demo page
- Scroll down to any alert with dismissible option (Dismissing demo or Dismiss on timeout)
- Find alert in DOM
- Dismiss alert (or wait time, after what it dismissed)
- Look at DOM again
- Add new alert
- Repeat steps 3-6 several times
alertnotremoved
multiplealertsatdom

### Expected result
All alerts after dismissing are removed from DOM

### Versions of ngx-bootstrap, Angular, and Bootstrap:

ngx-bootstrap: all supported

Angular: all supported

Bootstrap: all supported

### PR DESCRIPTION
**Pull request by IraErshova**
Fix removing alert from DOM after its dismissing

Close #3608

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+alerts/demos/dismiss-on-timeout/dismiss-on-timeout.html [hunk 1]
```diff
@@ -1,9 +1,5 @@
-<alert type="success" dismissOnTimeout="5000">
-  <strong>Well done!</strong> You successfully read this important alert message.
-</alert>
-
-<p>If you missed alert on top of me, just press <code>Add more</code> button</p>
+<p>If you missed alert under me, just press <code>Add more</code> button</p>
 <div *ngFor="let alert of alerts">
-  <alert [type]="alert.type" [dismissOnTimeout]="alert.timeout">{{ alert.msg }}</alert>
+  <alert [type]="alert.type" [dismissOnTimeout]="alert.timeout" (onClosed)="onClosed($event, alert)">{{ alert.msg }}</alert>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+alerts/demos/dismiss-on-timeout/dismiss-on-timeout.html]: I'm sure you can use just `onClosed(alert)`

File: demo/src/app/components/+alerts/demos/dismiss/dismiss.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <div *ngFor="let alert of alerts">
-  <alert [type]="alert.type" [dismissible]="dismissible">{{ alert.msg }}</alert>
+  <alert [type]="alert.type" [dismissible]="dismissible" (onClosed)="onClosed($event, alert)">{{ alert.msg }}</alert>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+alerts/demos/dismiss/dismiss.html]: same

File: demo/src/app/components/+typeahead/demos/async/async.ts [hunk 3]
```diff
@@ -16,11 +16,7 @@ export class DemoTypeaheadAsyncComponent {
   statesComplex: any[] = [
     { id: 1, name: 'Alabama', region: 'South' },
     { id: 2, name: 'Alaska', region: 'West' },
-    {
-      id: 3,
-      name: 'Arizona',
-      region: 'West'
-    },
+    { id: 3, name: 'Arizona', region: 'West' },
```
Comment (IlyaSurmay) [inline demo/src/app/components/+typeahead/demos/async/async.ts]: remove this file from commit

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Tested, looks good

File: demo/src/app/components/+alerts/demos/dismiss-on-timeout/dismiss-on-timeout.ts [hunk 4]
```diff
@@ -14,4 +18,8 @@ export class DemoAlertTimeoutComponent {
       timeout: 5000
     });
   }
+
+  onClosed(dismissedAlert: any): void {
```
Comment (valorkin) [inline demo/src/app/components/+alerts/demos/dismiss-on-timeout/dismiss-on-timeout.ts]: type param

File: src/alert/alert.component.ts [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
   Output
 } from '@angular/core';
 import { AlertConfig } from './alert.config';
-import { OnChange } from '../utils/decorators';
+import { OnChange } from '../utils';
```
Comment (valorkin) [inline src/alert/alert.component.ts]: this will break system.js

Comment (valorkin) [reply src/alert/alert.component.ts]: please revert

**Pull request merged**